### ISSUE TITLE
🐛 BUG: stopping unstable_dev workers throws ERR_SERVER_NOT_RUNNING

### ISSUE THREAD

**Issue by UnsungHero97** [cloudflare/workers-sdk#3335]
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version of `Wrangler` are you using?

3.0.0

### What operating system are you using?

macOS 13.3.1

### Describe the Bug

jest v29.5.0
typescript v5.0.0
wrangler v3.0.0

I'm running into an issue with stopping `unstable_dev` workers in my unit tests (jest). From what I can tell so far, it seems to be related to tests/workers running in parallel because the issue goes away if I run jest with `--runInBand`.

Here's the smallest setup to reproduce my issue:

```
// src/index.ts
export default {
  async fetch(request) {
    if (request.url.includes('test')) {
      return new Response(null, { status: 400 });
    }
    return new Response(null, { status: 404 });
  },
};

// src/api1.test.ts
describe('api1', () => {
  let worker;

  beforeAll(async () => {
    worker = await unstable_dev('src/index.ts', {
      experimental: { disableExperimentalWarning: true },
      local: true,
    });
  });

  afterAll(async () => {
      await worker.stop();
  });

  test('GET /', async () => {
    const response = await worker.fetch();
    expect(response.status).toEqual(404);
  });
});


// src/api2.test.ts
describe('api2', () => {
  let worker;

  beforeAll(async () => {
    worker = await unstable_dev('src/index.ts', {
      experimental: { disableExperimentalWarning: true },
      local: true,
    });
  });

  afterAll(async () => {
      await worker.stop();
  });

  test('POST /test', async () => {
    const response = await worker.fetch('/test', { method: 'POST' });
    expect(response.status).toEqual(400);
  });
});
```

<img width="250" alt="Screenshot 2023-05-23 at 4 45 41 PM" src="https://github.com/cloudflare/workers-sdk/assets/440299/d1ff9b85-5fc5-494e-919a-7f284658d2c8">
<img width="250" alt="Screenshot 2023-05-23 at 4 46 04 PM" src="https://github.com/cloudflare/workers-sdk/assets/440299/b7994b6c-8464-4169-9e18-fe61c4791a3e">

Wrapping `await worker.stop()` in a try-catch, the error is more clear:

<img width="400" alt="Screenshot 2023-05-23 at 4 48 05 PM" src="https://github.com/cloudflare/workers-sdk/assets/440299/4a0fee86-d33a-4a2e-b264-bba9307475ed">
<img width="400" alt="Screenshot 2023-05-23 at 4 47 20 PM" src="https://github.com/cloudflare/workers-sdk/assets/440299/177a7f45-f20e-4d50-a4b9-dbec7affc060">

Running jest in watch mode and hitting enter to trigger a new run will sometimes run cleanly and sometimes error; the error happens intermittently and nondeterministically without any changes to the code, sometimes `api1` fails and sometimes `api2` fails.

Aside from `--runInBand`, I haven't been able to find any other clues as to what might be happening.

**Comment by vassbence** [comment]
+1 facing the same issue with vitest

**Comment by mrbbot** [comment]
Hey @UnsungHero97! Thanks again for flagging this. In an effort to remove stale issues so we can have better signal on current friction points, we’re closing out issues that have not been updated within the last 6 months. Rahul has provided a workaround above, but if this is still a problem, please feel free to open a new issue referencing this one. Please be sure to include an updated summary and any additional context and we will review. Thank you!
